src/Auth.ts

```diff
@@ -199,46 +200,43 @@ export class Auth {
   }
 
   public async ensureAccessToken(resource: string, logger: Logger, debug: boolean = false, fetchNew: boolean = false): Promise<string> {
-    const now: Date = new Date();
```

Comment:
What if the specified access token expired? I suggest that we still perform this logic as early as possible to catch errors as soon as possible

Reply:
I didn't remove the logic, it's still there. also for the new authType